### PR TITLE
add postgress full text search operator

### PR DESCRIPTION
**Pull request by andrewthad**
work in progress

### CODE REVIEW

File: src/Database/Esqueleto/PostgreSQL.hs [hunk 1]
```diff
@@ -143,6 +144,14 @@ stringAgg ::
   -> SqlExpr (Value (Maybe s)) -- ^ Concatenation.
 stringAgg expr delim = stringAggWith AggModeAll expr delim []
```
Comment (parsonsmatt) [inline src/Database/Esqueleto/PostgreSQL.hs]:
```suggestion
-- | Full text search operator.
--
-- @since 2.8.0
```

**Comment by andrewthad** [comment]
Although this currently works correctly, it’s probably better to have the second argument to this function accept something different.  Maybe an algebraic data types for constructing  various full text search predicates.

**Comment by parsonsmatt** [comment]
Good point. Looks like it supports `&`, `|`, and `!`. It should be relatively straightforward to write that.

We may also want to incorporate more options for the `tsvector` type, too: https://www.postgresql.org/docs/9.5/datatype-textsearch.html 

It may be reasonable to have `@@.` for the simple case and `+@@.` (where `+` represents "extra") that's typed like `(+@@.) :: SqlExpr (Value TsVector) -> SqlExpr (Value TsQuery) -> SqlExpr Bool`.

**Comment by bitemyapp** [comment]
Gonna close this for now, I'd be happy to merge a complete version of this if you pick this back up and take it home.

**Pull request closed without merging**